frontend/src/gamestate.js

```diff
@@ -187,27 +191,84 @@ class GameState extends EventEmitter {
       landDistribution: this.#landDistribution,
       autopickCardIds: this.#autopickCardIds,
       picksPerPack: this.#picksPerPack,
+      burnCardIds: this.#burnCardIds
+    });
+  }
+
+  updPickState() {
+    this.emit("pickState", {
+      autopicks: this.#autopickCardIds,
+      burns: this.#burnCardIds
     });
   }
 
   isAutopick(cardId) {
     return this.#autopickCardIds.includes(cardId.toString());
   }
 
+  isBurn(cardId) {
+    return this.#burnCardIds.includes(cardId.toString());
+  }
+
   isAutoremovableAutopick(cardId, picksPerPack) {
     if (this.#autopickCardIds.length == picksPerPack && picksPerPack > 1){
       return this.#autopickCardIds[0] == cardId.toString();
     }
   }
   updateAutopick(cardId, picksPerPack) {
-    if (this.#autopickCardIds.length == picksPerPack) this.#autopickCardIds.shift();
+    if (this.#autopickCardIds.length == picksPerPack) {
+      this.#autopickCardIds.shift();
+    }
+
+    if (this.isBurn(cardId)) {
+      remove(this.#burnCardIds, id => id === cardId);
+    }
+
     this.#autopickCardIds.push(cardId);
     this.updState();
+    this.updPickState();
   }
 
   resetPack() {
     this.get(ZONE_PACK).length = 0;
     this.#autopickCardIds = [];
+    this.#burnCardIds = [];
+  }
+
+  addBurnCard(cardId, burnsPerPack) {
+    if (burnsPerPack <= 0) {
+      return false;
+    }
+
+    if (this.#burnCardIds.length == burnsPerPack) {
+      this.#burnCardIds.shift();
+    }
+
+    if (this.isAutopick(cardId)) {
+      remove(this.#autopickCardIds, id => id === cardId);
+    }
+
+    this.#burnCardIds.push(cardId);
+    this.updState();
+    this.updPickState();
+  }
+
+  isPickReady(picksPerPack, burnsPerPack) {
+    const packLength = this.get(ZONE_PACK).length;
+
+    if (packLength === (this.#autopickCardIds.length + this.#burnCardIds.length)) {
+      return true;
+    }
+
+    if (picksPerPack != this.#autopickCardIds.length) {
+      return false;
+    }
+
+    if (burnsPerPack != this.#burnCardIds.length) {
```

Comment:
Recommend using `! ==` in case for some reason an `undefined` gets compared to a 0 because of a programmer error in future